addons/website_sale/static/src/js/website_sale_tour_shop_cart_recovery.js

```diff
@@ -0,0 +1,104 @@
+odoo.define('website_sale.tour_shop_cart_recovery', function (require) {
+    'use strict';
+
+    var base = require('web_editor.base');
+    var localStorage = require('web.local_storage');
+    var rpc = require('web.rpc');
+    var tour = require('web_tour.tour');
+
+    var productName = "Acoustic Bloc Screens";
+    var orderIdKey = 'website_sale.tour_shop_cart_recovery.orderId';
+    var recoveryLinkKey = 'website_sale.tour_shop_cart_recovery.recoveryLink';
+
+    tour.register('shop_cart_recovery', {
+        test: true,
+        url: '/shop?search=' + productName,
+        wait_for: base.ready(),
+    },
+    [
+        {
+            content: "select " + productName,
+            trigger: '.oe_product_cart a:containsExact("' + productName + '")',
+        },
+        {
+            content: "click add to cart",
+            trigger: '#product_details #add_to_cart',
+        },
+        {
+            content: "check product is in cart, and get cart id",
+            trigger: 'td.td-product_name:contains(' + productName + ')',
+            run: function () {
+                rpc.query({
+                    'model': 'website',
+                    'method': 'sale_get_order',
+                    'args': [[$('html').data('website-id')]],
+                }).then(function (data) {
+                    var orderId = parseInt(data.match(/sale\.order\((\d+),\)/)[1]);
+                    localStorage.setItem(orderIdKey, orderId);
+                    $('body').append('<div id="shop_cart_recovery_rpc_done"></div>');
+                });
+            },
+        },
+        {
+            content: "wait for rpc to be done and logout",
+            trigger: '#shop_cart_recovery_rpc_done',
+            run: function () {
+                window.location.href = "/web/session/logout?redirect=/";
+            },
+        },
+        {
+            content: "go to login",
+            trigger: 'a[href="/web/login"]',
+        },
+        {
+            content: "login as admin and go to the SO (backend)",
+            trigger: '.oe_login_form',
+            run: function () {
+                var orderId = localStorage.getItem(orderIdKey);
+                var url = "/web#id=" + orderId + "&action=sale.action_orders&view_type=form";
+                $('.oe_login_form input[name="login"]').val("admin");
+                $('.oe_login_form input[name="password"]').val("admin");
+                $('.oe_login_form input[name="redirect"]').val(url);
+                $('.oe_login_form').submit();
```

Comment:
Suppose you could save the .oe_login_form jQuery element in a variable

Reply:
It was copied from another tour but ok, better with a variable.